docs/web/docs/explanations/architect_api.md

```diff
@@ -0,0 +1,108 @@
+
+The core carrier for information in the Mephisto Architect API is the `Packet` class. Downstream, they act as the way to tie an `Agent` class in python to an actual human worker. 
+
+The `Channel` is the primary way of trasmitting packets, with the `WebsocketChannel` being the main implementation Mephisto currently uses with its `Architect`s. The `ClientIOHandler` is responsible for using and interpreting packets, so it defines the key types to be handled.
+
+### Packet Types
+There are a number of different packet types used by Mephisto:
+- `PACKET_TYPE_ALIVE`: Used to mark the success of a new socket connection.
+- `PACKET_TYPE_SUBMIT`: Used to handle submission of any subcomponent of a `Unit`, currently either onboarding or full task. 
+- `PACKET_TYPE_OUTGOING_LIVE_DATA`: Used to send any new live data to an agent.
+- `PACKET_TYPE_INCOMING_LIVE_DATA`: Used for the frontend to send any type of data to the backend, usually to be processed by a user-defined callback. 
+- `PACKET_TYPE_REGISTER_WORKER`: Used to request a new `Agent` and `Unit` data for a specific worker on a task.
+- `PACKET_TYPE_AGENT_DETAILS`: Used to respond with the details of a worker registration request.
+- `PACKET_TYPE_REQUEST_STATUSES`: Used by Mephisto to poll for the current statuses for a worker.
+- `PACKET_TYPE_RETURN_STATUES`: Used by the router to return updates for all of the currently registered agents.
+- `PACKET_TYPE_ERROR`: Used by the router and frontend to communicate to the python backend that an error has occurred.
+
+### Architect Responsibilities
+
+While this is the "Architect API" most of the responsibilities for the architect are merely pointing the `ClientIOHandler` to the correct `Channel`s for sending packets for a given client. Ultimately it is the `ClientIOHandler` that dictates the responsibilities that the transmitted messages carry:
+- The `ClientIOHandler` must send a `PACKET_TYPE_ALIVE` whenever it opens a new channel (in this case, to a router).
+- For `Unit` Registration, in response to a `PACKET_TYPE_REGISTER_WORKER`, the handler must return a `PACKET_TYPE_AGENT_DETAILS` with the details of an agent and it's initialization data, or the failure status for why an agent couldn't be created.
+- During a `Unit` the handler must process `PACKET_TYPE_INCOMING_LIVE_DATA` and direct the content to the correct handlers, and should send a `PACKET_TYPE_OUTGOING_LIVE_DATA` for any `Agent.send_data()` call on a live connected `Agent`. The handler must also process `PACKET_TYPE_SUBMIT` packets for the key transitions of a `Unit` in progress, and should respond with `PACKET_TYPE_AGENT_DETAILS` for a submit on an `OnboardingAgent`.
+- Over any run, the handler should poll with `PACKET_TYPE_REQUEST_STATUS` and update local `Agent` statuses on disconnects from `PACKET_TYPE_RETURN_STATUS`. This also acts as a heartbeat from the Python core to the router. The handler should also take `PACKET_TYPE_ERROR` and log the contents if this ever occurs.
```

Comment:
[Nit]

It says _STATUS (singular) here whereas above it says _STATUSES (plural)